compiler/AST/AggregateType.cpp

```diff
@@ -875,6 +893,19 @@ static void resolveConcreteFields(AggregateType* ret, CallExpr* call, const char
   }
 }
 
+static Symbol* substitutionForField(Symbol* field, SymbolMap& subs) {
+  Symbol* retval = NULL;
+
+  form_Map(SymbolMapElem, e, subs) {
+    if (field->name == e->key->name) {
```

Comment:
Yes, it was using strcmp before, but it doesn't need to because all of the strings involved are astrs. Symbol::verify already checks that the name is an astr.